virtool/groups/data.py

```diff
@@ -36,12 +37,17 @@ async def get(self, group_id: str) -> VTGroup:
         """
         document = await self._db.groups.find_one(group_id)
 
+        user_list = [
+            UserNested(**base_processor(user))
+            async for user in self._db.users.find({"groups": {"$in": [group_id]}})
```

Comment:
This should work just as well.
```suggestion
            async for user in self._db.users.find({"groups": group_id})
```
You don't actually have to use the `$in` operator. Sorry for the misdirection.

---

What you do here is identical to the code in `update`.

Extract to a function in `virtool.groups.db` and call it `fetch_group_users()`. It will just need to take a `db`  and `group_id` and return the list of nested users.